src/validation.cpp

```diff
@@ -3396,22 +3396,21 @@ bool IsWitnessEnabled(const CBlockIndex* pindexPrev, const Consensus::Params& pa
 
 int GetWitnessCommitmentIndex(const CBlock& block)
 {
-    int commitpos = -1;
     if (!block.vtx.empty()) {
-        for (size_t o = 0; o < block.vtx[0]->vout.size(); o++) {
-            const CTxOut& vout = block.vtx[0]->vout[o];
+        for (size_t o = block.vtx[0]->vout.size(); o > 0;) {
+            const CTxOut& vout = block.vtx[0]->vout[--o];
```

Comment:
This 1-based to 0-based array indexing conversion within the body seems rather odd, why not simply 0-base it right from the start:

```cpp
        for (int o = block.vtx[0]->vout.size()-1; o >= 0; o--) {
            const TxOut& vout = block.vtx[0]->vout[o];
```

Drawback is that the loop counter type has to be changed from `size_t` (unsigned) to `int` signed, but I don't see why that would be a problem.

Reply:
Yes, need to change to signed. I'm fine with both TBH. Let's see what others say.